server/db/dbManager.ts

```diff
@@ -53,22 +61,24 @@ export class PostManager {
 		if (pageSize * (pageNum - 1) > count) {
 			return [];
 		}
-		let sorted;
+		let sorted = this.model.find({ parent: undefined });
+		if (search !== '') {
```

Comment:
This simple check also tracks an empty string.

```suggestion
		if (search) {
```